app/models/cart.rb

```diff
@@ -10,4 +11,12 @@ def update_total
         end
         total
     end
+
+    def calculate_quantity
+        total = 0
```

Comment:
rename this var to quantity or total_quantity, as you are in the model it may be saving the calculation on the total attribute of the record